engine/core/src/main/java/org/datacleaner/reference/regexswap/RegexSwapClient.java

```diff
@@ -19,150 +19,101 @@
-            final Node categoriesNode = categoriesNodes.get(0);
-            final List<Node> categoryNodes = HttpXmlUtils.getChildNodesByName(categoriesNode, "category");
-            for (final Node categoryNode : categoryNodes) {
-                final String categoryName = HttpXmlUtils.getText(categoryNode);
-                final Category category = getCategoryByName(categoryName);
-                if (category != null) {
-                    categories.add(category);
-                }
-            }
-        }
-        final Regex regex =
-                new Regex(name, description, expression, author, timestamp, positiveVotes, negativeVotes, detailsUrl,
-                        categories);
-        _regexes.put(name, regex);
-        return regex;
-    }
-
-    public Regex refreshRegex(Regex regex) {
-        final String detailsUrl = regex.getDetailsUrl();
-        final Element regexNode = HttpXmlUtils.getRootNode(_httpClient, detailsUrl);
-        regex = createRegex(regexNode);
-        return regex;
-    }
-
-    public List<Regex> getRegexes(final Category category) {
-        final List<Regex> regexes = new ArrayList<>();
-        final Node rootNode = HttpXmlUtils.getRootNode(_httpClient, category.getDetailsUrl());
-        final List<Node> regexNodes = HttpXmlUtils.getChildNodesByName(rootNode, "regex");
-        for (final Node regexNode : regexNodes) {
-
-            final String name = HttpXmlUtils.getChildNodeText(regexNode, "name");
-            final String description = HttpXmlUtils.getChildNodeText(regexNode, "description");
-            final String expression = HttpXmlUtils.getChildNodeText(regexNode, "expression");
-            final String author = HttpXmlUtils.getChildNodeText(regexNode, "author");
-            final long timestamp = Long.parseLong(HttpXmlUtils.getChildNodeText(regexNode, "timestamp"));
-            final int positiveVotes = Integer.parseInt(HttpXmlUtils.getChildNodeText(regexNode, "positiveVotes"));
-            final int negativeVotes = Integer.parseInt(HttpXmlUtils.getChildNodeText(regexNode, "negativeVotes"));
-            final String detailsUrl = HttpXmlUtils.getChildNodeText(regexNode, "detailsUrl");
-
-            final List<Category> categories;
-            Regex regex = _regexes.get(name);
-            if (regex == null) {
-                categories = new ArrayList<>();
-                regex = new Regex(name, description, expression, author, timestamp, positiveVotes, negativeVotes,
-                        detailsUrl, categories);
-            } else {
-                categories = regex.getCategories();
-                if (!categories.contains(category)) {
-                    categories.add(category);
-                }
-            }
-
-            _regexes.put(name, regex);
-
-            regexes.add(regex);
-        }
-        return regexes;
-    }
+	private static final URI REGEXES_URI = URI.create("https://datacleaner.github.io/content/regexes.json");
```

Comment:
Leaving the URL as a hardcoded value may leave us open to a similar issue again later.

Reply:
Hmm yes but I find it hard to imagine a situation where we wouldn't have to pull a release to fix it anyway?
Is your suggestion that we use an environment variable or something like that?

Reply:
Yeah, just some kind of fallback. I guess an environment variable that overrides the default would be perfect for that.

Just so that someone who doesn't want to update for compatibility or whatever reason (or who wants a custom set of regexes hosted locally), can update the URL.

But its no biggie.

Reply:
Good point, fixed.